server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2286,6 +2295,9 @@ public SegmentInfos getLatestSegmentInfos() {
         OpenSearchDirectoryReader reader = null;
         try {
             reader = externalReaderManager.internalReaderManager.acquire();
+            if (engineConfig.isReadOnly()) {
+                return ((StandardDirectoryReader)((SoftDeletesDirectoryReaderWrapper) reader.getDelegate()).getDelegate()).getSegmentInfos();
```

Comment:
Can you please add a comment here that this is safe because we always wrap the Standard reader with a `SoftDeletesDirectoryReaderWrapper ` for replicas when segrep is on.  This will be more clear once we have a separate engine class.